digitalforms-api/src/main/java/ca/bc/gov/open/pssg/rsbc/digitalforms/controller/DisclosureController.java

```diff
@@ -0,0 +1,104 @@
+import org.springframework.web.bind.annotation.RestController;
+
+import ca.bc.gov.open.pssg.rsbc.digitalforms.model.JSONResponse;
+import ca.bc.gov.open.pssg.rsbc.digitalforms.model.DisclosureWrapper;
+import ca.bc.gov.open.pssg.rsbc.digitalforms.model.DocumentWrapper;
+import ca.bc.gov.open.pssg.rsbc.digitalforms.service.DisclosureService;
+import ca.bc.gov.open.pssg.rsbc.digitalforms.util.DigitalFormsConstants;
+import ca.bc.gov.open.pssg.rsbc.digitalforms.util.DigitalFormsUtils;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+/**
+ * 
+ * Disclosure service Controller.
+ * 
+ * @author sivakaruna
+ *
+ */
+@RestController
+@Api(value = "Disclosure", tags = { "Disclosure" })
+public class DisclosureController {
+
+	private class DisclosureDocumentSwaggerResponse extends JSONResponse<DocumentWrapper> {
+	}
+
+	private class DocumentDisclosedSwaggerResponse extends JSONResponse<Boolean> {
+	}
+
+	@Autowired
+	DisclosureService service;
+
+	private final Logger logger = LoggerFactory.getLogger(DisclosureController.class);
+
+	@GetMapping(value = { "**/disclosure/**",
+			"/{documentId}/disclosure/{correlationId}" }, produces = DigitalFormsConstants.JSON_CONTENT)
+	@ApiOperation(value = "Get disclosure document", response = JSONResponse.class)
+	@ApiResponses(value = {
+			@ApiResponse(code = 200, message = "Success", response = DisclosureDocumentSwaggerResponse.class) })
+	public ResponseEntity<JSONResponse<DocumentWrapper>> disclosureDocumentGet(
+			@PathVariable(value = "documentId", required = true) String documentId,
+			@PathVariable(value = "correlationId", required = true) String correlationId) {
+
+		MDC.put(DigitalFormsConstants.REQUEST_CORRELATION_ID, correlationId);
+		MDC.put(DigitalFormsConstants.REQUEST_ENDPOINT, "disclosureDocumentGet");
+		logger.info("Get disclosure document request received");
+
+		// TODO Update based on ORDS service
+		JSONResponse<DocumentWrapper> data = service.getDisclosureDocument(documentId, correlationId);
+		if (data != null) {
+			logger.info("Get disclosure document request success");
+			MDC.clear();
+			return new ResponseEntity<>(data, HttpStatus.OK);
+		} else {
+			logger.info("Get disclosure document data not found");
+			MDC.clear();
+			return new ResponseEntity<>(
+					DigitalFormsUtils.buildErrorResponse(DigitalFormsConstants.NOT_FOUND_ERROR, 404),
+					HttpStatus.NOT_FOUND);
+		}
+	}
+
+	@PatchMapping(value = { "**/disclosure/**",
+			"/{noticeNumber}/disclosure/{correlationId}" }, consumes = DigitalFormsConstants.JSON_CONTENT, produces = DigitalFormsConstants.JSON_CONTENT)
+	@ApiOperation(value = "Set Disclosure Document Sent", response = JSONResponse.class)
+	@ApiResponses(value = {
+			@ApiResponse(code = 200, message = "Success", response = DocumentDisclosedSwaggerResponse.class) })
+	public ResponseEntity<JSONResponse<Boolean>> disclosureDocumentPatch(
```

Comment:
Can we update the name of the method here for clarity. Maybe "markDisclosureSent" or something similar. Also clarify the logging message too.  This will make it easier for new devs to understand what's going on here. Thanks!

Reply:
updated method name to setDisclosureSent 
updated logger statements